### PR TITLE
Use default ENV variables when a task is triggered from API

### PR DESCRIPTION
**Pull request by kirs**
I've noticed that when the restart task is triggered by API with a `spy` command, the deploy script doesn't get the default ENV variables, unlike when the task is triggered from the browser.

IMO, when you declare something like:

```
tasks:
  restart:
    action: "Restart Application"
    steps:
      - bin/cap $ENVIRONMENT deploy:restart --trace
    variables:
      -
        name: MYVARIABLE
        title: blahblah
        default: "1"
```

You rely on the fact that `MYVARIABLE` is always present, not only when the task is triggered from the web.

This PR fixes the edge case and extends the test coverage.

review @DazWorrall @casperisfine

### CODE REVIEW

File: test/dummy/db/schema.rb [hunk 1]
```diff
@@ -189,24 +189,24 @@
   end
 
   create_table "tasks", force: :cascade do |t|
-    t.integer  "stack_id",              limit: 4,                         null: false
-    t.integer  "since_commit_id",       limit: 4,                         null: false
-    t.integer  "until_commit_id",       limit: 4,                         null: false
-    t.string   "status",                limit: 10,    default: "pending", null: false
+    t.integer  "stack_id",              limit: 4,                            null: false
```
Comment (casperisfine) [inline test/dummy/db/schema.rb]: You are not supposed to have any schema change.

File: app/models/shipit/task_definition.rb [hunk 2]
```diff
@@ -48,6 +48,10 @@ def filter_envs(env)
       EnvironmentVariables.with(env).permit(variables)
     end
 
+    def variables_with_defaults
+      @variables_with_defaults ||= variables.select { |v| v.default.present? }
```
Comment (casperisfine) [inline app/models/shipit/task_definition.rb]: What if the default is a blank string?

```ruby
>> ' '.present?
=> false
```

Comment (kirs) [inline app/models/shipit/task_definition.rb]: I don't think that someone will add a default ENV variable with `' '` as a default value, but I think we can stop doing `to_s` in [models/shipit/variable_definition.rb](https://github.com/Shopify/shipit-engine/blob/master/app/models/shipit/variable_definition.rb#L8) and we'll know when the default value is `nil`. I'll update the PR.

**Comment by kirs** [comment]
Updated the PR 👌
